pkg/obfuscate/obfuscate.go

```diff
@@ -304,8 +307,17 @@ func NewObfuscator(cfg Config) *Obfuscator {
 	if cfg.Logger == nil {
 		cfg.Logger = noopLogger{}
 	}
+	optsStr := ""
+	optsBytes, err := json.Marshal(cfg.SQL)
+	if err == nil {
+		optsStr = string(optsBytes)
+	} else {
+		log.Errorf("failed to marshal obfuscation config: %v", err)
+	}
+
 	o := Obfuscator{
 		opts:              &cfg,
+		sqlOptsStr:        optsStr,
```

Comment:
Any particular reason to cache `optsStr` and not `z.MemHashString(optsStr)` directly ?